tools/generate_chain.py

```diff
@@ -1,25 +1,27 @@
 import cProfile
```

Comment:
another minor thing, but some commands seem to have executable permissions while others like `generate_chain.py` don't

Reply:
I run all the python programs with an explicit call to `python`. setting the executable bit would also have required a `#!` comment at the top I believe